iris-location-service/src/main/java/iris/location_service/service/LocationService.java

```diff
@@ -29,20 +30,47 @@
 
 	private final @NotNull DBSearchIndex index;
 
-	public void addLocations(String providerId, List<LocationInformation> locations) {
-		// TODO: Authenticate API Access
+	static boolean isValidAndNotNullEmail(String email) {
+		if (email == null)
+			return false;
+		String regex = "^[\\w-_\\.+]*[\\w-_\\.]\\@([\\w]+\\.)+[\\w]+[\\w]$";
+		return email.matches(regex);
+	}
 
-		// TODO: Define sensible limits for this API
+	static boolean isValidAndNotNullPhoneNumber(String phoneNumber) {
+		if (phoneNumber == null)
+			return false;
+		String regex =
+			"^(\\+\\d{1,3}( )?)?(\\d{3}[ ]?){2}\\d{3}$" + "|^((((\\()?0\\d{4}(\\))?)|(\\+49 (\\()?\\d{4}(\\))?))([/ -])(\\d{6}(-\\d{2})?))$"
+				+ "|^(\\+\\d{1,3}( )?)?((\\(\\d{3}\\))|\\d{3})[- .]?\\d{3}[- .]?\\d{4}$" + "|^(\\+\\d{1,3}( )?)?(\\d{3}[ ]?)(\\d{2}[ ]?){2}\\d{2}$";
```

Comment:
I fear we are too strict with this regex. I think we should write some dedicated unit tests (also for email) and maybe put the functions to a separate validator class.

Reply:
I have adjusted the regex based on the tests. I think the regex allows for a lot of different formats already. If we find something we want to add that can be done later.